MediaGraph/topologies/evr-motion-files-assets/readme.md

```diff
@@ -1,8 +1,8 @@
-# Event-based video recording to local files and Assets based on motion events
+# Event-based video recording to local files and assets based on motion events
 
-This topology enables you perform event-based recording. The video from an RTSP-capable camera is analyzed for the presence of motion. When motion is detected, events are sent to a signal gate processor node which opens, sending frames to a file sink node. As a result, new files (MP4 format) are created on the local file system of the Edge device, containing the frames where motion was detected. In addition, the motion events are used to trigger the signal gate processor node which will send frames to the asset sink node when motion is present. As a result, new Assets are created containing clips where motion was detected.
+This topology enables you to record motion-detected video clip to the edge and the cloud. The video from an RTSP-capable camera is analyzed for the presence of motion. When motion is detected, events are sent to a signal gate processor node which opens, sending frames to a file sink as well as asset sink. As a result of the file sink, new files (MP4 format) are created on the local file system of the edge device, containing the frames where motion was detected. In addition as a result of the asset sink connected to the same signal gate, corresponding assets with same content as edge mp4 files are created on the cloud.
```

Comment:
This topology enables you perform event-based recording of video clips to the cloud as well as to the edge. The video from an RTSP-capable camera is analyzed for the presence of motion. When motion is detected, events are sent to a signal gate processor node which opens, sending frames to a file sink node as well as an asset sink node. As a result, new files (MP4 format) are created on the local file system of the Edge device, and new assets are created in your Azure Media Service account in the cloud. The recordings contain the frames where motion was detected. 

Note: the topology creates new MP4 files each time motion is detected. Over time, this can fill up the local filesystem. You should monitor the contents of the output directory and prune older files as necessary.

Reply:
changed